server/src-lib/Hasura/GraphQL/Execute/LiveQuery/Plan.hs

```diff
@@ -264,76 +270,109 @@ data ReusableLiveQueryPlan
+-- here. We should evaluate if we need it here.
 buildLiveQueryPlan
   :: ( MonadError QErr m
-     , MonadReader r m
-     , Has UserInfo r
-     , Has FieldMap r
-     , Has OrdByCtx r
-     , Has QueryCtxMap r
-     , Has SQLGenCtx r
-     , Has (L.Logger L.Hasura) r
      , MonadIO m
-     , Tracing.MonadTrace m
-     , HasVersion
      )
-  => E.Environment
-  -> PGExecCtx
-  -> QueryReusability
-  -> QueryActionExecuter
-  -> ObjectSelectionSet
+  => PGExecCtx
+  -> UserInfo
+  -> InsOrdHashMap G.Name (SubscriptionRootField UnpreparedValue)
   -> m (LiveQueryPlan, Maybe ReusableLiveQueryPlan)
-buildLiveQueryPlan env pgExecCtx initialReusability actionExecuter selectionSet = do
-  ((resolvedASTMap, (queryVariableValues, syntheticVariableValues)), finalReusability) <-
-    runReusabilityTWith initialReusability $
-      flip runStateT mempty $ flip OMap.traverseWithKey (unAliasedFields $ unObjectSelectionSet selectionSet) $
-      \_ field -> case GV._fName field of
-        "__typename" -> throwVE "you cannot create a subscription on '__typename' field"
-        _ -> do
-          unresolvedAST <- GR.queryFldToPGAST env field actionExecuter
-          resolvedAST <- GR.traverseQueryRootFldAST resolveMultiplexedValue unresolvedAST
-
-          let (_, remoteJoins) = GR.toPGQuery resolvedAST
-          -- Reject remote relationships in subscription live query
-          when (remoteJoins /= mempty) $
-               throw400 NotSupported
-                       "Remote relationships are not allowed in subscriptions"
-          pure resolvedAST
-
-  userInfo <- asks getter
-  let multiplexedQuery = mkMultiplexedQuery resolvedASTMap
+buildLiveQueryPlan pgExecCtx userInfo unpreparedAST = do
+  -- ((resolvedASTs, (queryVariableValues, syntheticVariableValues)), finalReusability) <-
+  --   GV.runReusabilityTWith initialReusability . flip runStateT mempty $
+  --     fmap Map.fromList . for (toList fields) $ \field -> case GV._fName field of
+  --       "__typename" -> throwVE "you cannot create a subscription on '__typename' field"
+  --       _ -> do
+  --         unresolvedAST <- GR.queryFldToPGAST field actionExecutioner
+  --         resolvedAST <- GR.traverseQueryRootFldAST resolveMultiplexedValue unresolvedAST
+
+  --         let (_, remoteJoins) = GR.toPGQuery resolvedAST
+  --         -- Reject remote relationships in subscription live query
+  --         when (remoteJoins /= mempty) $
+  --              throw400 NotSupported
+  --                      "Remote relationships are not allowed in subscriptions"
+  --         pure (GV._fAlias field, resolvedAST)
+
+  -- Transform the RQL AST into a prepared SQL query
+{-  preparedAST <- for unpreparedAST \unpreparedQuery -> do
+    (preparedQuery, PlanningSt _ planVars planVals)
+      <- flip runStateT initPlanningSt
+      $  traverseSubscriptionRootField prepareWithPlan unpreparedQuery
+    pure $! irToRootFieldPlan planVars planVals preparedQuery
+-}
```

Comment:
Is this because we don't do query caching for now?
Should we add a `-- See Note [Temporarily disabling query plan caching]` here too?

Reply:
There's a bunch of "old" code commented out here that's _replaced_ by code below. So at least some code can be removed here.